libs/utils/utils.js

```diff
@@ -638,17 +638,36 @@ const decorateCopyLink = (a, evt) => {
   });
 };
 
+function convertStageLinks(config, anchors) {
+  if (config.env?.name === 'prod' || !config.stageDomainsMap) return;
+  [...anchors].forEach((a) => {
+    const { hostname } = window.location;
+    const matchedRules = Object.entries(config.stageDomainsMap)
+      .find(([domain]) => hostname.includes(domain));
+
+    if (matchedRules) {
+      const [, domainsMap] = matchedRules;
+      const matchedDomain = Object.keys(domainsMap)
+        .find((domain) => a.href.includes(domain));
+
+      if (matchedDomain) {
+        a.href = a.href.replace(a.hostname, domainsMap[matchedDomain] === 'origin'
+          ? hostname
+          : domainsMap[matchedDomain]);
+      }
+    }
```

Comment:
Could we get even greedier with early returns?
```suggestion
    if (!matchedRules) return;
    const [, domainsMap] = matchedRules;
    const matchedDomain = Object.keys(domainsMap)
      .find((domain) => a.href.includes(domain));

    if (!matchedDomain) return;
    a.href = a.href.replace(a.hostname, domainsMap[matchedDomain] === 'origin'
      ? hostname
      : domainsMap[matchedDomain]);
```